Functional.Primitives.Extensions/ResultExtensions.cs

```diff
@@ -129,5 +129,32 @@ public static class ResultExtensions
 
 		public static async Task<Result<TSuccess, TFailure>> FailureIfNone<TSuccess, TFailure>(this Task<Result<Option<TSuccess>, TFailure>> result, Func<TFailure> failureFactory)
 			=> (await result).FailureIfNone(failureFactory);
+
+		public static Result<TSuccess, TFailure> Try<TInputSuccess, TSuccess, TFailure>(this Result<TInputSuccess, TFailure> result, Func<TInputSuccess, TSuccess> successFactory, Func<Exception, TFailure> failureFactory)
```

Comment:
What about TrySelect?

Reply:
Changed to `TrySelect` as this is a combination of both Try and Select.